In [27]:
HTML(read(open("style.css"), String))

HTML{String}("<link href='http://fonts.googleapis.com/css?family=Alegreya+Sans:100,300,400,500,700,800,900,100italic,300italic,400italic,500italic,700italic,800italic,900italic' rel='stylesheet' type='text/css'>\r\n<link href='http://fonts.googleapis.com/css?family=Arvo:400,700,400italic' rel='stylesheet' type='text/css'>\r\n<link href='http://fonts.googleapis.com/css?family=PT+Mono' rel='stylesheet' type='text/css'>\r\n<link href='http://fonts.googleapis.com/css?family=Shadows+Into+Light' rel='stylesheet' type='text/css'>\r\n<link href='http://fonts.googleapis.com/css?family=Philosopher:400,700,400italic,700italic' rel='stylesheet' type='text/css'>\r\n\r\n<style>\r\n\r\n@font-face {\r\n    font-family: \"Computer Modern\";\r\n    src: url('http://mirrors.ctan.org/fonts/cm-unicode/fonts/otf/cmunss.otf');\r\n}\r\n\r\n.container { width: 100% }\r\n\r\n/* Formatting for header cells */\r\n.text_cell_render h1 {\r\n    font-family: 'Philosopher', sans-serif;\r\n    font-weight: 400;\r\n    font-size: 2.2em;\r\n    line-height: 100%;\r\n    color: rgb(0, 80, 120);\r\n    margin-bottom: 0.1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\t\r\n.text_cell_render h2 {\r\n    font-family: 'Philosopher', serif;\r\n    font-weight: 400;\r\n    font-size: 1.9em;\r\n    line-height: 100%;\r\n    color: rgb(200,100,0);\r\n    margin-bottom: 0.1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\t\r\n\r\n.text_cell_render h3 {\r\n    font-family: 'Philosopher', serif;\r\n    margin-top:12px;\r\n    margin-bottom: 3px;\r\n    font-style: italic;\r\n    color: rgb(94,127,192);\r\n}\r\n\r\n.text_cell_render h4 {\r\n    font-family: 'Philosopher', serif;\r\n}\r\n\r\n.text_cell_render h5 {\r\n    font-family: 'Alegreya Sans', sans-serif;\r\n    font-weight: 300;\r\n    font-size: 16pt;\r\n    color: grey;\r\n    font-style: italic;\r\n    margin-bottom: .1em;\r\n    margin-top: 0.1em;\r\n    display: block;\r\n}\r\n\r\n.text_cell_render h6 {\r\n    font-family: 'PT Mono', sans-serif;\r\n    font-weight: 300;\r\n    font-size: 10pt;\r\n    color: grey;\r\n    margin-bottom: 1px;\r\n    margin-top: 1px;\r\n}\r\n\r\n.text_cell_render em {\r\n    font-family: 'Philosopher', sans-serif;\r\n    color:        blue;\r\n    background-color: rgb(255,220,180);\r\n    font-size:    110%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   100;\r\n}\r\n\r\n.text_cell_render b {\r\n    color:            rgb(255,195,195);\r\n    background-color: rgb(0,0,0);\r\n    font-size:    110%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   650;\r\n}\r\n\r\n.text_cell_render tt {\r\n    font-size:    120%;\r\n    margin-left:   2px;\r\n    margin-right:  2px;\r\n    font-weight:   150;\r\n}\r\n\r\n.Codemirror {\r\n    font-family: \"PT Mono\";\r\n    font-size: 100%;\r\n}\r\n\r\n</style>\r\n\r\n")

# Quiescence Search Testing

In [28]:
using Pkg
# Pkg.add("NBInclude")
using NBInclude
# Pkg.add("BenchmarkTools")
using BenchmarkTools

In [29]:
@nbinclude("QuiescenceSearch.ipynb")

strategicQS (generic function with 1 method)

In [30]:
@nbinclude("IterativeDeepening.ipynb")

iterativeDeepening (generic function with 1 method)

In [31]:
zobrist = generate_zobrist_hashing()

ZobristHashing(UInt64[0x1a861418e6008a62 0x93a64dee689272f1 … 0x7b26b6513376e601 0x78c3c0b6f2f6dea9; 0x3b7270bd58226dea 0x527d0bb8e1f92703 … 0x3c0c89cbae1a6976 0x41e8ae508db9ded7; … ; 0x27efcbda7ef5fb34 0x408b33024cc457ee … 0x8ac1c447842f72c1 0x042bc97beca90cfd; 0x36816819f98b7e26 0xe2d09b1c53cdbfce … 0x8c8f915e70797545 0xc6717cd733119a63], UInt64[0xabc4674a1f4dfb7e, 0x4cd3fba62abbe731, 0x55242739249d668b, 0x1398714056bdd97b], UInt64[0x673763bc98160f2a, 0x66cc8380d96cbba9, 0xb6291868eab41b1a, 0x71db1ec220058bba, 0xf3a90ced7c85e983, 0x46582f82b3bdba11, 0xbe4c8bf82870cbf5, 0x108d3de967f260da], 0xb7a893e36e518b57)

## Benchmarking tests

In [32]:
b1 = fromfen("rn3rk1/pbppq1pp/1p2pb2/4N2Q/3PN3/3B4/PPP2PPP/R3K2R w KQ - 7 11")
display(b1)
@btime see(b1, movefromstring("h5e8"))  #, movefromstring("h5e8")

Board (rn3rk1/pbppq1pp/1p2pb2/4N2Q/3PN3/3B4/PPP2PPP/R3K2R w KQ -):
 r  n  -  -  -  r  k  - 
 p  b  p  p  q  -  p  p 
 -  p  -  -  p  b  -  - 
 -  -  -  -  N  -  -  Q 
 -  -  -  P  N  -  -  - 
 -  -  -  B  -  -  -  - 
 P  P  P  -  -  P  P  P 
 R  -  -  -  K  -  -  R

  236.199 ns (4 allocations: 560 bytes)


-10

Benchmark results

1. chess.jl see function 241.727 ns
1. isTactical 213.452 ns
1. isQuiet 1610.000 ns
    1. 345ns durch moves(State)

In [34]:
b2 = fromfen("r1b1k2r/pp1nqpp1/4p2p/3pP1N1/8/3BQ3/PP3PPP/2R2RK1 w kq - 0 1")
display(b2)
@btime noQuiesce(b2, evaluate_position(b2))
# @btime quiesceSee(b2, evaluate_position(b2), -Inf, Inf)
# @btime quiesce(b2, evaluate_position(b2), zobrist_hash(b2), 0, -Inf, Inf)

Board (r1b1k2r/pp1nqpp1/4p2p/3pP1N1/8/3BQ3/PP3PPP/2R2RK1 w kq -):
 r  -  b  -  k  -  -  r 
 p  p  -  n  q  p  p  - 
 -  -  -  -  p  -  -  p 
 -  -  -  p  P  -  N  - 
 -  -  -  -  -  -  -  - 
 -  -  -  B  Q  -  -  - 
 P  P  -  -  -  P  P  P 
 -  -  R  -  -  R  K  -

  21.700 μs (316 allocations: 25.88 KiB)


-40

Benchmark results of b2 = fromfen("r1b1k2r/pp1nqpp1/4p2p/3pP1N1/8/3BQ3/PP3PPP/2R2RK1 w kq - 0 1")
1. noQuiesce   21.700 μs (316 allocations: 25.88 KiB) => -40
1. quiesceSee  27.000 μs (391 allocations: 43.20 KiB) => -40
1. quiesce     53.400 μs (690 allocations: 45.05 KiB) => -40